### PR TITLE
[SP-15] - Fetch Concordia API

### LINKED ISSUE #20: Academic calendar events
Description:
As a user, I want Concordia's undergraduate academic dates to be integrated in my calendar so I can keep track of important dates (eg. add/drop, disc, etc).

The user will have Concordia’s undergraduate academic dates integrated in their calendars so they can stay on track with important deadlines.
All dates
Business Value: L
Priority: high
Effort: 13

### PR DESCRIPTION
**Pull request by jacobmathewlopez**
Closes #20 
- Implemented service with authentication to access Concordia Schedule Data
  - Authenticated through [Concordia Open Data](https://opendata.concordia.ca/admin/index.php#)
  - Uses Access-Control-Allow-Origin to proxy requests to allow 3rd party API
  - Can be used by calling endpoint in the FE like this:
  
```
axios.get(`${process.env.REACT_APP_BASE_URL}openData/schedule/*/BIOL/201`, { mode: 'cors' }).then(scheduleData => {
         console.log(`scheduleData`,scheduleData)
     }).catch(e => {
         console.log(`Error: ${e}`)
})
```

### CODE REVIEW

File: src/routes/openData.js
Comment (mnashat) [inline src/routes/openData.js:28]: API credentials should not be left in the code. Please move to local .env file.

**Review by mnashat — CHANGES_REQUESTED** [review]
In addition to my comment about API credentials, we should coordinate before merging as I am also working on ConU Open Data API and have a opendata.js route. We should merge these together before merging to main. Also, please check sonarcloud errors and fix accordingly. Thank you!

**Comment by guaj** [comment]
the goal here was to get a concept of proof (working code that fetch concordia api). As @jacobmathewlopez mentionned, there are multiple ways we can implement this in our system. Needs analysis. Will not accept this PR.

**Comment by jacobmathewlopez** [comment]
@guaj Sorry your comment is a little hard to understand.. This code successfully fetches current Concordia schedules and serves as an initial layout for accessing Concordia Open Data. 

- Requests must be done in BE to avoid cross origin errors.
- One implementation to get this information is to use an route to call open data API from FE to use information on FE (this is what I am doing currently, since I assumed you needed this information).
- Another solution could be only using the openData.js to gather the data, then use this response to create new academic events all done in BE. 
- Let me know which solution works for you guys, really depends on if FE team needs the data returned by Concordia.

File: src/routes/openData.js
Comment (jacobmathewlopez) [reply src/routes/openData.js:28]: No problem man, let me know when you are ready to merge and we can pair to implement a final structure for Concordia Open Data :)

Comment (jacobmathewlopez) [reply src/routes/openData.js:28]: I think I am going to close this PR? I have just reviewed your branch data_fetch_ConU_open_data, and it seems you have already done the route for academic events and openData connection in the best way possible!

**Comment by guaj** [comment]
> @guaj Sorry your comment is a little hard to understand.. This code successfully fetches current Concordia schedules and serves as an initial layout for accessing Concordia Open Data.
> 
> * Requests must be done in BE to avoid cross origin errors.
> * One implementation to get this information is to use an route to call open data API from FE to use information on FE (this is what I am doing currently, since I assumed you needed this information).
> * Another solution could be only using the openData.js to gather the data, then use this response to create new academic events all done in BE.
> * Let me know which solution works for you guys, really depends on if FE team needs the data returned by Concordia.

Hey sorry, what I meant is that we should not merge this branch in main if we are not using it right now. We can leave this as a branch for now.

**Comment by mnashat** [comment]
Is this PR still required/relevant?

**Comment by mnashat** [comment]
This pull request is stale. If required, it can be reopened later.

**Pull request closed without merging**
